### DISCUSSION TITLE
Let us import via clients or disable https requirement to use webvault

### DISCUSSION THREAD

**Discussion by Snazzie** [dani-garcia/vaultwarden#2274] [Ideas]
Bitwarden doesnt let us import unless its through the web vault. Bitwarden doesnt let us use the webvault unless we have https set up.

I dont want to set up https.
please let us import via the client we're using, or allow us to disable the https requirement.

**Comment by RealOrangeOne** [comment]
This unfortunately is a restriction of your browser, not bitwarden, let along vaultwarden. A lot of the new fancy crypto functionality in browsers requires using a HTTPS connection.

Side note: HTTPS for a password manager is probably a good idea anyway.

**Comment by BlackDex** [comment]
If you really do not want to use https you should make it so that you can access it via localhost, that is the only exception for this encryption api's.

**Comment by Snazzie** [comment]
True.

But regardless if im using http or https, i should be able to import via the client since the client allows adding new entries, to which i will relay to the client repo.

**Reply by RealOrangeOne** [reply]
> regardless if im using http or https, i should be able to import via the client

Still, no, I'm afraid not. Importing requires the vault. The vault requires crypto, and crypto requires HTTPS.

https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto

**Comment by yllekz** [comment]
So just to clarify, in order to use this application internally on an LAN we need to spin up an entire PKI for this?

**Reply by BlackDex** [reply]
If you want to access the application via the web-vault and somehow can not access it via a localhost/127.0.0.1 connection, then you need some form of SSL Certificate to get it working in current browsers yes.

If you do not want to to use SSL for the web-vault, either use haproxy to proxy a localhost connection to the actual vaultwarden instance.
Or use something like `ssh -L 127.0.0.1:8080:127.0.0.1:8080 vaultwarden-host`.

The client apps work just fine without SSL, but the web-vault not, unless accessed on localhost, that is not something we or Bitwarden can fix or change, since that is a protection from the browsers.

**Reply by RealOrangeOne** [reply]
In short, yes. Self-signed certificates will also work though.

Your options are either to have it accessible over `https` (with whatever certificate), or have your browser be able to access it via `localhost` (or similar). Nothing (bit|vault)warden can do about that.
